### ISSUE TITLE
[Checkbox] Accesibility requirements

### ISSUE THREAD

**Issue by aweell** [dxc-technology/halstack-style-guide#166]
[WAI-ARIA Authoring Practices 1.2](https://www.w3.org/TR/wai-aria-practices-1.2/#checkbox) for `checkbox` component

### Keyboard Interaction
* When the checkbox has focus, pressing the Space key changes the state of the checkbox :white_check_mark:

### Roles, States, and Properties

* The checkbox has role checkbox :white_check_mark:
* The checkbox has an accessible label provided by one of the following:
   * Visible text content contained within the element with role checkbox.
   * A visible label referenced by the value of aria-labelledby set on the element with role checkbox.
  aria-label set on the element with role checkbox.
* When checked, the checkbox element has state aria-checked set to true.
* When not checked, it has state aria-checked set to false.

[WAI-ARIA Checkbox example](https://www.w3.org/TR/wai-aria-practices/examples/checkbox/checkbox-1/checkbox-1.html)

- [x] Checkbox component guidelines and implementation should be updated to cover all the accessibility recommendations

**Comment by aweell** [comment]
Closed via #207
